opm/parser/eclipse/EclipseState/Grid/GridProperties.hpp

```diff
@@ -124,6 +146,17 @@ namespace Opm {
         void handleCOPYREGRecord( const DeckRecord& record, const GridProperty<int>& regionProperty );
 
 
+        struct iterator : public storageIteratorType {
+            iterator( storageIteratorType iter ) : storageIteratorType( iter ) { }
+
+            const GridProperty<T>& operator*( ) {
+                const auto& pair = storageIteratorType::operator*( );
+                return pair.second;
```

Comment:
Well - readability is in the mind of the beholder; I prefer the one with the temporary.